v1/edgeloadbalancer/policies_http_types_matches.go

```diff
@@ -0,0 +1,297 @@
+	PoliciesHTTPClientIPMatch struct {
+		// Criteria to use for IP address matching the HTTP request.
+		// var PoliciesHTTPClientIPMatchCriteria and PoliciesHTTPClientIPMatchCriteriaString are defined get the list of valid values.
+		Criteria string `validate:"required,oneof=IS_IN IS_NOT_IN"`
+		// Either a single IP address, a range of IP addresses or a network CIDR. Must contain at least
+		// one item.
+		Addresses []string `validate:"required,dive,ipv4|cidr|ipv4_range"`
+	}
+
+	PoliciesHTTPServicePortMatch struct {
+		// Criteria to use for port matching the HTTP request.
+		// var PoliciesHTTPServicePortMatchCriteria and PoliciesHTTPServicePortMatchCriteriaString are defined get the list of valid values.
+		Criteria string `validate:"required,oneof=IS_IN IS_NOT_IN"`
+		// Listening TCP ports.
+		Ports []int `validate:"required,dive,tcp_udp_port"`
+	}
+
+	PoliciesHTTPMethodMatch struct {
+		// Criteria to use for HTTP method matching the HTTP request.
+		// var PoliciesHTTPMethodMatchCriteria and PoliciesHTTPMethodMatchCriteriaString are defined get the list of valid values.
+		Criteria string `validate:"required,oneof=IS_IN IS_NOT_IN"`
+		// HTTP methods.
+		// var PoliciesHTTPMethodsMatch and PoliciesHTTPMethodsMatchString are defined get the list of valid values.
+		Methods []string `validate:"required,dive,oneof=GET POST PUT DELETE PATCH OPTIONS TRACE CONNECT PROPFIND PROPPATCH MKCOL COPY MOVE LOCK UNLOCK"`
+	}
+
+	PoliciesHTTPPathMatch struct {
+		// Criteria to use for path matching the HTTP request.
+		// var PoliciesHTTPPathMatchCriteria and PoliciesHTTPPathMatchCriteriaString are defined get the list of valid values.
+		Criteria string `validate:"required,oneof=BEGINS_WITH DOES_NOT_BEGIN_WITH CONTAINS DOES_NOT_CONTAIN ENDS_WITH DOES_NOT_END_WITH EQUALS DOES_NOT_EQUAL REGEX_MATCH REGEX_DOES_NOT_MATCH"`
+		// String values to match the path
+		MatchStrings []string `validate:"required"`
+	}
+
+	PoliciesHTTPHeadersMatch []PoliciesHTTPHeaderMatch
+
+	PoliciesHTTPHeaderMatch struct {
+		// Criteria to use for header matching the HTTP request.
+		// var PoliciesHTTPHeaderMatchCriteria and PoliciesHTTPHeaderMatchCriteriaString are defined get the list of valid values.
+		Criteria string `validate:"required,oneof=BEGINS_WITH DOES_NOT_BEGIN_WITH CONTAINS DOES_NOT_CONTAIN ENDS_WITH DOES_NOT_END_WITH EQUALS DOES_NOT_EQUAL EXISTS DOES_NOT_EXIST"`
+		// Name of the HTTP header whose value is to be matched
+		Name string `validate:"required"`
+		// String values to match for an HTTP header
+		Values []string `validate:"required|excluded_if=Criteria EXISTS DOES_NOT_EXIST"`
+	}
+
+	PoliciesHTTPCookieMatch struct {
+		// Criteria to use for cookie matching the HTTP request.
+		// var PoliciesHTTPCookieMatchCriteria and PoliciesHTTPCookieMatchCriteriaString are defined get the list of valid values.
+		Criteria string `validate:"required,oneof=BEGINS_WITH DOES_NOT_BEGIN_WITH CONTAINS DOES_NOT_CONTAIN ENDS_WITH DOES_NOT_END_WITH EQUALS DOES_NOT_EQUAL"`
```

Comment:
```suggestion
		Criteria string `validate:"required,oneof=BEGINS_WITH DOES_NOT_BEGIN_WITH CONTAINS DOES_NOT_CONTAIN ENDS_WITH DOES_NOT_END_WITH EQUALS DOES_NOT_EQUAL EXISTS DOES_NOT_EXIST"`
```
or remove the excluded in Value